Gems/ROS2/Code/Source/Lidar/ROS2LidarSensorComponent.cpp

```diff
@@ -154,31 +159,116 @@ namespace ROS2
         {
             point = inverseLidarTM.TransformPoint(point);
         }
-
-        auto* ros2Frame = GetEntity()->FindComponent<ROS2FrameComponent>();
+        auto* ros2Frame = Utils::GetGameOrEditorComponent<ROS2FrameComponent>(GetEntity());
```

Comment:
@PawelLiberadzki I think that class channels should no be published by default. The number of use cases is limited and publication of empty data makes recorded bags and throughput larger.

Reply:
This is a bit of an discussion question, in theory colors and class ids are not necessary and in theory there could be some toggle to turn this feature / off

Reply:
I agree with @Fireronin , it should be optional.